azavea/iow-boundary-tool
# Draw Map User Interface

## Overview
This PR adds the UI elements to the Draw page map. It also adds some basic polygon functionality (without using a global store/state).

Closes #24 

### Notes

As of now, the polygon edits don't persist when toggling visibility/edit mode. This is because there is no global store for this data.

## To Do
As part of this PR:
- [ ] Center modal/dialogs over map, rather than whole screen
- [x] Prevent double click from recentering map when clicking on other elements

Possibly a future PR:
- [ ] Search features
- [ ] Basemap and basemap layers

Definitely a future PR:
- [ ] Add state/store so polygon data will persist. This will require a bit of refactoring.

## Testing Instructions

- Go to http://localhost:4545/draw
- Click "Draw Polygon": a square should appear in the center of the map
- Click "New Polygon": a modal should pop up that allows you to change the name of the polygon
- Click "Show/Hide": this should toggle the visibility of the polygon
- Click "Edit Points": this should toggle whether the point drag handles are visible and usable
- Click "Delete Polygon": after confirmation, this should delete the polygon
- The polygon should be editable by dragging midpoints to create new verticies, or dragging corners to move them.
- The "Zoom In" and "Zoom Out" buttons should zoom in and out respectively

## Checklist
